src/main/java/org/web3j/tx/QuorumRawTransactionManager.java

```diff
@@ -0,0 +1,90 @@
+package org.web3j.tx;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
+import okhttp3.RequestBody;
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.RawTransaction;
+import org.web3j.crypto.TransactionEncoder;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.Web3jService;
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.methods.response.EthSendTransaction;
+import org.web3j.quorum.methods.request.PrivateTransaction;
+import org.web3j.utils.Numeric;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+
+public class QuorumRawTransactionManager extends RawTransactionManager {
+
+    private Web3jService web3jService;
+
+    private List<String> privateFor;
+
+    private OkHttpClient httpClient;
+
+    private String thirdPartyUrl;
+
+    public QuorumRawTransactionManager(OkHttpClient httpClient, String thirdPartyUrl, Web3j web3j, Web3jService web3jService,
+                                       Credentials credentials, List<String> privateFor, int attempts,
+                                       int sleepDuration) {
+        super(web3j, credentials, attempts, sleepDuration);
+        this.web3jService = web3jService;
+        this.httpClient = httpClient;
+        this.privateFor = privateFor;
+        this.thirdPartyUrl = thirdPartyUrl;
+    }
+
+
+    @Override
+    public EthSendTransaction signAndSend(RawTransaction rawTransaction) throws IOException {
+
+        if (null != privateFor && privateFor.size() > 0) {
+            rawTransaction = storeRaw(rawTransaction);
+        }
+
+        byte[] signedMessage = TransactionEncoder.signMessage(rawTransaction, credentials);
+
+        String hexValue = Numeric.toHexString(signedMessage);
+
+        PrivateTransaction privateTransaction = new PrivateTransaction(null,
+                null, null, null, null, null, null, privateFor);
+
+        return new Request<>(
+                "eth_sendRawPrivateTransaction",
```

Comment:
this seems to be a new RPC API in Quorum, can you link the PR in Quorum for this new API?

Reply:
Nam just raised the [pull request](https://github.com/jpmorganchase/quorum/pull/591).

Thanks for the refactoring.

When the blk.io team merge their [PR ](https://github.com/web3j/quorum/pull/16) we should be able to remove the QuromuRawTransactionManager and use their version.